### PR TITLE
Refactored GUI to the Nth Degree

### PR DESCRIPTION
**Pull request by brianseidl**
# US10008
- Added lobby page where user can choose if they want to play against chungus or multiplayer mode
- Refactored GUI to use routes and views instead of having everything in public
- Made a file `big_chungus.py` to be the AI but currently contains random agent script
- When game ends user gets redirected to lobby page
- Dealer server is now a child process and no longer needs to be run separately when running server
- Feel like I'm missing something here

## FYI
- You might need to run `npm install` again
- To run server, now run `node bin/www` from `/acpc-gui-client`

Have fun reviewing this @gclark422

### CODE REVIEW

File: acpc-python-client/human_agent.py [hunk 1]
```diff
@@ -84,10 +84,10 @@ def _data_serializer(self, game, match_state, is_acting_player):
...
 
         data = {
-        'is_acting_player': is_acting_player,
-        'round': current_round,
-        'board_cards': board_cards,
-        'pot': pot,
+            'is_acting_player': is_acting_player,
+            'round': current_round,
+            'board_cards': board_cards,
+            'pot': pot,
```
Comment (brianseidl) [inline acpc-python-client/human_agent.py]: I was looking at this and I was like, "Why does this look so weird?" The indentation is off.

File: acpc-gui-client/bin/www [hunk 2]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
```
Comment (brianseidl) [inline acpc-gui-client/bin/www]: this whole file is auto generated. That's why I used the IDE.

File: acpc-gui-client/public/javascripts/sketch.js [hunk 3]
```diff
@@ -51,62 +51,62 @@ function setup() {
...
-        43: loadImage("img/queen_of_spades.png"),
-        44: loadImage("img/king_of_clubs.png"),
-        45: loadImage("img/king_of_diamonds.png"),
-        46: loadImage("img/king_of_hearts.png"),
-        47: loadImage("img/king_of_spades.png"),
-        48: loadImage("img/ace_of_clubs.png"),
-        49: loadImage("img/ace_of_diamonds.png"),
-        50: loadImage("img/ace_of_hearts.png"),
-        51: loadImage("img/ace_of_spades.png"),
+        0: loadImage("images/2_of_clubs.png"),
```
Comment (gclark422) [inline acpc-gui-client/public/javascripts/sketch.js]: was this necessary? lol

File: acpc-gui-client/routes/game.js [hunk 4]
```diff
@@ -1,21 +1,17 @@
 var express = require('express');
+var router = express.Router();
 var {PythonShell} = require('python-shell');
-var spawn = require('child_process').spawn;
 var socket = require('socket.io');
-var app = express();
-var server = app.listen(3000);
 
-app.use(express.static('public'));
+var io = socket(6969);
```
Comment (gclark422) [inline acpc-gui-client/routes/game.js]: _Nice._

File: acpc-gui-client/public/javascripts/sketch.js (hunk 3 above)
Comment (brianseidl) [reply acpc-gui-client/public/javascripts/sketch.js]: I think I fixed that issue in a later commit.

**Pull request merged**